examples/environment.py

```diff
@@ -44,8 +46,39 @@ def before_all(context: Context):
             sys.exit(1)
 
     context.model_graph = g
-    before_all_isaac(context=context, headless=True, time_step_sec=DEFAULT_ISAAC_PHYSICS_DT_SEC)
-
+    
+    getattr(context, "use_livestream", False)
+    getattr(context, "headless", True)
+    getattr(context, "render", False)
+    if context.use_livestream:
+        getattr(context, "width", 1280)
+        getattr(context, "height", 720)
+        getattr(context, "window_width", 1920)
+        getattr(context, "window_height", 1080)
+        getattr(context, "hide_ui", False)
+        getattr(context, "renderer", "RayTracedLighting")
+        getattr(context, "display_options", 3286)
+        getattr(context, "draw_mouse", True)
+        getattr(context, "protocol", "ws")
+        getattr(context, "enable_nginx", False)
+        assert context.render is True, "Livestream mode requires rendering. Set 'render' to True."
+        assert context.headless is True, "Livestream mode requires headless mode. Set 'headless' to True."
+        
+    before_all_isaac(context=context, headless=context.headless, time_step_sec=DEFAULT_ISAAC_PHYSICS_DT_SEC)
+
+def read_config_file(context, filename="config.yaml"):
```

Comment:
Default value for `filename` is not necessary, you only use it once. Also add type annotations pls.